zebra-chain/src/parameters/network.rs

```diff
@@ -53,21 +50,109 @@ const ZIP_212_GRACE_PERIOD_DURATION: HeightDiff = 32_256;
 
 /// An enum describing the possible network choices.
 #[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Hash, Serialize, Deserialize)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
 pub enum Network {
     /// The production mainnet.
     #[default]
     Mainnet,
 
     /// The oldest public test network.
-    Testnet,
+    Testnet(#[serde(skip)] TestnetParameters),
+}
+
+/// Configures testnet network parameters to use instead of default values.
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Hash, Serialize)]
+pub struct TestnetParameters(Option<NetworkParameters>);
+
+impl std::ops::Deref for TestnetParameters {
+    type Target = Option<NetworkParameters>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<Option<NetworkParameters>> for TestnetParameters {
+    fn from(value: Option<NetworkParameters>) -> Self {
+        Self(value)
+    }
+}
+
+impl TestnetParameters {
+    /// Returns `network_id` in network parameters, if any.
+    pub fn network_id(self) -> Option<[u8; 4]> {
+        self.0.and_then(NetworkParameters::network_id)
+    }
+
+    /// Returns `default_port` in network parameters, if any.
+    pub fn default_port(self) -> Option<u16> {
+        self.0.and_then(NetworkParameters::default_port)
+    }
+
+    /// Returns `cache_name` in network parameters, if any.
+    pub fn cache_name(self) -> Option<&'static str> {
+        self.0.and_then(NetworkParameters::cache_name)
+    }
+
+    /// Returns `genesis_hash` in network parameters, if any.
+    pub fn genesis_hash(self) -> Option<&'static str> {
+        self.0.and_then(NetworkParameters::genesis_hash)
+    }
+
+    /// Returns `activation_heights` in network parameters, if any.
+    pub fn activation_heights(self) -> Option<&'static [(Height, NetworkUpgrade)]> {
+        self.0.and_then(NetworkParameters::activation_heights)
+    }
+}
+
+/// Configures network parameters to use instead of default values.
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Hash, Serialize)]
+pub struct NetworkParameters {
+    /// Used as the network magic, Zebra will reject messages and connections from peers
+    /// with a different network magic.
+    network_id: Option<[u8; 4]>,
```

Comment:
This is going to make the code a bit easier to read, maybe we could move the type from `zebra-network` to `zebra-chain` in a PR, then base this PR on it?

```suggestion
    network_magic: Option<network::Magic>,
```